native/java/org/jpype/classloader/DynamicClassLoader.java

```diff
@@ -25,16 +27,69 @@
 import java.util.Set;
 import java.util.jar.JarEntry;
 import java.util.jar.JarFile;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
-public class DynamicClassLoader extends ClassLoader
+public class DynamicClassLoader extends URLClassLoader
 {
 
   List<URLClassLoader> loaders = new LinkedList<>();
   HashMap<String, ArrayList<URL>> map = new HashMap<>();
 
   public DynamicClassLoader(ClassLoader parent)
   {
-    super(parent);
+    super(launch(), parent);
+  }
+
+  private static URL[] launch()
+  {
+    String cp = System.getProperty("jpype.class.path");
+    if (cp == null)
+      return new URL[0];
+
+    ArrayList<URL> path = new ArrayList<>();
+    int off = 0, next;
+    do
+    {
+      next = cp.indexOf(File.pathSeparator, off);
+      String element = (next == -1)
+              ? cp.substring(off)
+              : cp.substring(off, next);
+      if (!element.isEmpty())
+      {
+        try
+        {
+          URL url = Paths.get(element).toUri().toURL();
+          if (url != null)
+            path.add(url);
+        } catch (MalformedURLException ex)
+        {
+          System.err.println("Malformed url "+ element);
+        } catch (IOException ex)
+        {
+          System.err.println("Unable to open "+ element);
+        }
+      }
+      off = next + 1;
+    } while (next != -1);
+
+    System.out.println("jpype.class.path " + cp);
+    System.clearProperty("jpype.class.path");
+    System.setProperty("java.class.path", cp);
+    return path.toArray(new URL[0]);
+  }
```

Comment:
@tachyonicClock This section allows for us to load the non-ascii path just like a normal one.  Thus removing the need for late load logic.